core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -675,7 +675,7 @@ class DAGScheduler(
     // Cancel all jobs belonging to this job group.
     // First finds all active jobs with this group id, and then kill stages for them.
     val activeInGroup = activeJobs.filter(activeJob =>
-      groupId == activeJob.properties.get(SparkContext.SPARK_JOB_GROUP_ID))
+      Option(activeJob.properties).exists(_.get(SparkContext.SPARK_JOB_GROUP_ID) == groupId))
```

Comment:
Now that we've changed `initialValue`, won't `activeJob.properties` always be non-null now?  I think it should now be safe to use the old code, unless I'm overlooking something.